lib/foreman_maintain/concerns/system_helpers.rb

```diff
@@ -92,6 +92,14 @@ def execute(command, options = {})
       def shellescape(string)
         Shellwords.escape(string)
       end
+
+      def fetch_etc_hostname
+        execute('hostname')
+      end
+
+      def fetch_server_ip
+        execute("ip route get 8.8.8.8 | awk 'NR==1 {print $NF}'").strip
```

Comment:
It seems like this could depend on the system having access to internet to work properly.